search/tasks_test.py

```diff
@@ -40,3 +56,180 @@ def test_increment_document_integer_field_task(mocked_api):
     increment_document_integer_field(*indexing_api_args)
     assert mocked_api.increment_document_integer_field.call_count == 1
     assert mocked_api.increment_document_integer_field.call_args[0] == indexing_api_args
+
+
+def test_wrap_retry_exception():
+    """wrap_retry_exception should raise RetryException when other exceptions are raised"""
+    with wrap_retry_exception(KeyError):
+        # Should not raise an exception
+        pass
+
+
+@pytest.mark.parametrize("matching", [True, False])
+def test_wrap_retry_exception_matching(matching):
+    """A matching exception should raise a RetryException"""
+
+    class SubError(KeyError):
+        """Use a subclass to assert isinstance use"""
+
+    def raise_thing():
+        """raise the exception"""
+        if matching:
+            raise SubError()
+        else:
+            raise TabError()
+
+    matching_exception = RetryException if matching else TabError
+    with pytest.raises(matching_exception):
+        with wrap_retry_exception(KeyError):
+            raise_thing()
+
+
+def test_index_post(mocker):
+    """index_post should call the api function of the same name"""
+    index_post_mock = mocker.patch('search.indexing_api.index_post')
+    wrap_mock = mocker.patch('search.tasks.wrap_retry_exception')
+    api_username = 'username'
+    post_id = 'post_id'
+    index_post.delay(api_username, post_id)
+
+    index_post_mock.assert_called_once_with(api_username, post_id)
+    wrap_mock.assert_called_once_with(PrawcoreException, PRAWException)
+
+
+def test_index_channel(mocker):
+    """index_channel should all posts of a channel"""
+    user = UserFactory.create()
+    index_post_mock = mocker.patch('search.tasks.index_post', autospec=True)
+    api_mock = mocker.patch('channels.api.Api', autospec=True)
+
+    expected_exception = ZeroDivisionError
+
+    replace_mock = mocker.patch('celery.app.task.Task.replace', return_value=expected_exception)
+    group_mock = mocker.patch('search.tasks.group', autospec=True)
+    get_channel_mock = api_mock.return_value.get_channel
+    post1 = mocker.Mock(id=1)
+    post2 = mocker.Mock(id=2)
+    get_channel_mock.return_value.new.return_value = [
+        post1,
+        post2,
+    ]
+    wrap_mock = mocker.patch('search.tasks.wrap_retry_exception')
+    channel_name = 'channel'
+    with pytest.raises(expected_exception):
+        index_channel.delay(user.username, channel_name)
+
+    api_mock.assert_called_once_with(user)
+    get_channel_mock.assert_called_once_with(channel_name)
+
+    wrap_mock.assert_called_once_with(PrawcoreException, PRAWException)
+    assert group_mock.call_count == 1
+    list(group_mock.call_args[0][0])  # iterate through generator
```

Comment:
Confused about this as well. Needs a more descriptive comment at the very least

Reply:
The `group(...)` which is called is actually operating on a generator, not a list. To assert the functions being called inside the generator we need to force iteration so that the calls are made. Do you think `list(...)` works for that here or should I do something else?